extensions/data-workspace/src/main.ts

```diff
@@ -16,6 +16,8 @@ import { IconPathHelper } from './common/iconHelper';
 export function activate(context: vscode.ExtensionContext): Promise<IExtension> {
 	const workspaceService = new WorkspaceService(context);
 	workspaceService.loadTempProjects();
+	workspaceService.checkForProjectsNotAddedToWorkspace();
```

Comment:
hmm, can you also try add a workspace folder, I am not sure whether that will cause a reload, if not, we have to have another event hook to do this check.

Reply:
good catch! Added an event hook to do the check for projects not in the workspace when the workspace folders change